subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -72,6 +72,36 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec implements
         mode << [0755, 0776]
     }
 
+    @Requires(UnitTestPreconditions.Symlinks)
+    def "symlinked file permissions are preserved in copy action"() {
+        given:
+        def testSourceFile = file(testFileName)
+        testSourceFile << "test file content"
+        testSourceFile.mode = mode
+
+        def testSourceFileLink = file("${testFileName}_link").createLink(testSourceFile.getRelativePathFromBase())
+
+        and:
+        buildFile << """
+        task copy(type: Copy) {
+            from "${testSourceFile.absolutePath}"
+            from "${testSourceFileLink.absolutePath}"
+            into ("build/tmp")
+        }
+        """
+
+        when:
+        run "copy"
+
+        then:
+        file("build/tmp/${testFileName}").mode == mode
+        file("build/tmp/${testFileName}_link").mode == mode
+
+        where:
+        mode << [0746, 0746]
```

Comment:
Nah, it's just badly formatted, I'll change it into table. See other tests.

Reply:
Or even remove it and hardcode the value in the test body, right?

Reply:
It might be still useful for adding more cases with other permissions, but yeah, let's just hardcode it and hope this problem wouldn't manifest anymore:)